### DISCUSSION TITLE
WPF - Unit Testing

### DISCUSSION THREAD

**Discussion by navalkishorsah** [cefsharp/CefSharp#4638] [General]
When we attempt to open the WPF window as part of one of the unit tests, we receive the following error. The control used in this window is ChromiumWebBrowser. [CefSharp Version: 113.1.40]

Error: Exit code is -532462766 ([1115/161107.951:WARNING:chrome_browser_cloud_management_controller.cc(87)] Could not create policy manager as CBCM is not enabled.)

Would you kindly let us know how to solve this?

**Comment by amaitland** [comment]
> 1115/161107.951:WARNING:chrome_browser_cloud_management_controller.cc(87)] Could not create policy manager as CBCM is not enabled

It's just a warning and generally safe to ignore in my experience. Unlikely related to any problem you are having.

For `Unit` testing you need to make sure you:
- Only run `CefSharp` in the default `AppDomain` (it won't run in any other due to limitations of `Microsoft VC++`)
- Call `Cef.Initialzie` and `Cef.Shutdown` on the **SAME** thread.

I'd suggest using `xUnit` if you aren't already as it can support disabling `AppDomains` which you'll need. The `CefSharp.Test` project as a reference.

**Comment by navalkishorsah** [comment]
@amaitland 
Thank you for your response!

The NUnit Framework is being used. Despite having the necessary DLLs at the proper location, we are encountering the following exception when debugging the unit test.

System.IO.FileNotFoundException: 'Could not load file or assembly 'CefSharp, Version=113.1.40.0, Culture=neutral, PublicKeyToken=40c4b6fc221f4138' or one of its dependencies. The system cannot find the file specified.

This exception occurs at InitializeComponent() OR Cef.Initialize

Would you kindly recommend a fix for the NUnit framework?

**Reply by amaitland** [reply]
Here are two of the key parts of the `xUnit` project. 

- https://github.com/cefsharp/CefSharp/blob/master/CefSharp.Test/BindingRedirectAssemblyResolver.cs
- https://github.com/cefsharp/CefSharp/blob/master/CefSharp.Test/CefSharpFixture.cs

Make sure your tests are running in the default `AppDomain`.

> System.IO.FileNotFoundException: 'Could not load file or assembly 'CefSharp, Version=113.1.40.0, Culture=neutral, PublicKeyToken=40c4b6fc221f4138' or one of its dependencies. The system cannot find the file specified.

You might need an assembly resolver as per example linked above.
